### ISSUE TITLE
MPS ignores storage offset of zero-dimensional CPU tensor views in mixed-device arithmetic

### ISSUE THREAD

**Issue by dhruvaron** [pytorch/pytorch#193057]
### 🐛 Describe the bug

When a zero-dimensional CPU tensor view with a nonzero storage offset participates directly in arithmetic with an MPS tensor, the MPS result uses the first value in the underlying CPU storage rather than the value selected by the view.

Both operand orders are affected. Cloning the CPU scalar view, converting it to a Python float, or moving it to MPS produces the correct result.

### Reproduction

```python
import torch

assert torch.backends.mps.is_available()

num_steps = 2048
noise_max = 80.0
noise_min = 0.002
rho = 7.0

i = torch.arange(num_steps)
steps = (
    noise_max ** (1 / rho)
    + i
    * (
        noise_min ** (1 / rho)
        - noise_max ** (1 / rho)
    )
    / (num_steps - 2)
) ** rho
steps[-1] = 0

indices = [0, 100, 1600]

cases = [
    ("CPU view", lambda s: s, 1),
    ("CPU clone", lambda s: s.clone(), 2),
    ("Python float", lambda s: s.item(), 3),
    ("MPS tensor", lambda s: s.to("mps"), 4),
]

for label, prepare, size in cases:
    x = torch.ones(size, device="mps")
    print(f"\n{label}")

    for index in indices:
        expected = steps[index].item()
        scalar = prepare(steps[index])

        left = (scalar * x).cpu()[0].item()
        right = (x * scalar).cpu()[0].item()

        print(
            f"index={index:4d} expected={expected:12.8f} "
            f"scalar*x={left:12.8f} x*scalar={right:12.8f}"
        )
```

### Actual result

```text
CPU view
index=   0 expected= 79.99998474 scalar*x= 79.99998474 x*scalar= 79.99998474
index= 100 expected= 60.94487000 scalar*x= 79.99998474 x*scalar= 79.99998474
index=1600 expected=  0.10995213 scalar*x= 79.99998474 x*scalar= 79.99998474

CPU clone
index=   0 expected= 79.99998474 scalar*x= 79.99998474 x*scalar= 79.99998474
index= 100 expected= 60.94487000 scalar*x= 60.94487000 x*scalar= 60.94487000
index=1600 expected=  0.10995213 scalar*x=  0.10995213 x*scalar=  0.10995213

Python float
index=   0 expected= 79.99998474 scalar*x= 79.99998474 x*scalar= 79.99998474
index= 100 expected= 60.94487000 scalar*x= 60.94487000 x*scalar= 60.94487000
index=1600 expected=  0.10995213 scalar*x=  0.10995213 x*scalar=  0.10995213

MPS tensor
index=   0 expected= 79.99998474 scalar*x= 79.99998474 x*scalar= 79.99998474
index= 100 expected= 60.94487000 scalar*x= 60.94487000 x*scalar= 60.94487000
index=1600 expected=  0.10995213 scalar*x=  0.10995213 x*scalar=  0.10995213
```

### Expected result

The `CPU view` results should equal the selected CPU view values, matching the cloned, Python-float, and MPS-tensor cases.

### Impact

This can silently corrupt iterative numerical algorithms. In the application where this was found, a timestep view that correctly printed approximately `0.1` was treated as `80` in mixed CPU/MPS multiplication, causing an RK integration trajectory to diverge by more than 30 orders of magnitude before overflowing. CUDA handled the same timestep view correctly.

### Workaround

Cloning the CPU scalar view before mixed MPS arithmetic works:

```python
scalar = steps[index].clone()
result = mps_tensor * scalar
```

Converting with `.item()` or transferring the scalar to MPS also works.

### Environment

- PyTorch: 2.11.0
- Python: 3.13.13
- macOS: 26.6.1 (build 25G76)
- Hardware: Apple M4 Pro
- `torch.backends.mps.is_built()`: `True`
- `torch.backends.mps.is_available()`: `True`

cc @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk @Isalia20

**Comment by DivitP** [comment]
This issue appears to be closely related to #187117, which describes the same storage_offset() issue for CPU 0-dim views used with MPS binary operations. The observed behavior and workarounds here seem to match that report.

**Comment by malfet** [comment]
Closing as duplicate of #187117, fixed in main and upcoming 2.14 release
